src/groups/validation_group.jl

```diff
@@ -0,0 +1,760 @@
+
+Identity(VG::ValidationLieGroup) = Identity(VG.lie_group)
+
+# similar to `internal_value` but just for Validation  types
+unwrap_validation(v) = v
+unwrap_validation(vTV::ValidationLieAlgebraTangentVector) = vTV.value
+unwrap_validation(vP::ValidationMPoint) = vP.value
+
+#
+#
+# An access helper function
+
+"""
+    _vLc(M::ValidationLieGroup, f::Function, context::Symbol)
+    _vLc(M::ValidationLieGroup, f::Function, context::NTuple{N,Symbol}) where {N}
+    _vLc(M::ValidationLieGroup, ::Nothing, context::NTuple{N,Symbol}) where {N}
+
+Return whether a check should be performed within `f` and the `context`(`s`) provided,
+if the second argument is `:Nothing`, only the context is checked
+
+This function returns false and hence indicates not to check, when
+* (one of the) `context`(`s`) is in the ignore list for `f` within `ignore_functions`
+* (one of the) `context`(`s`) is in the `ignore_contexts` list
+
+Otherwise the test is active.
+
+!!! Note
+   This function is internal and used very often, so it has a very short name;
+    `_vLc` stands for "`ValidationLieGroup` check".
+"""
+function _vLc end
+
+function _vLc(G::ValidationLieGroup, ::Nothing, context::Symbol)
+    # Similarly for the global contexts
+    (:All ∈ G.ignore_contexts) && return false
+    (context ∈ G.ignore_contexts) && return false
+    return true
+end
+function _vLc(G::ValidationLieGroup, f::Function, context::Symbol)
+    if haskey(G.ignore_functions, f)
+        # If :All is present -> deactivate
+        !_vLc(G.ignore_functions[f], :All) && return false
+        # If any of the provided contexts is present -> deactivate
+        !_vLc(G.ignore_functions[f], context) && return false
+    end
+    !_vLc(G, nothing, context) && return false
+    return true
+end
+function _vLc(G::ValidationLieGroup, f, contexts::NTuple{N,Symbol}) where {N}
+    for c in contexts
+        !_vLc(G, f, c) && return false
+    end
+    return true
+end
+# Sub tests: is any of a in b? Then return false – b is from before always a symbol already
+# If a and b are symbols, equality is checked
+_vLc(a::Symbol, b::Symbol) = !(a === b)
+# If a is a vector multiple, then return false if b appears in a
+_vLc(a::Union{<:NTuple{N,Symbol} where {N},<:AbstractVector{Symbol}}, b::Symbol) = !(b ∈ a)
+
+"""
+    _msg(G::ValidationLieGroup, str; error=:None, within::Union{Nothing,<:Function} = nothing,
+    context::Union{NTuple{N,Symbol} where N} = NTuple{0,Symbol}())
+
+issue a message `str` according to the mode `mode` (as `@error`, `@warn`, `@info`).
+"""
+function _msg(
+    G::ValidationLieGroup,
+    str;
+    error=G.mode,
+    within::Union{Nothing,<:Function}=nothing,
+    context::Union{NTuple{N,Symbol} where N}=NTuple{0,Symbol}(),
+)
+    !_vLc(G, within, context) && return nothing
+    (error === :error) && (throw(ErrorException(str)))
+    (error === :warn) && (@warn str)
+    (error === :info) && (@info str)
+    return nothing
+end
```

Comment:
Could you maybe reuse `_msg` from ManifoldsBase? If you extend `_vMc` instead of defining `_vLc` it should work fine I think?

Reply:
there was some reason why I had to rewrite `_vMc` to `_vLc`, the `_msg` I then rewrote to not depend on ManifoldsBase internals – though that is of course more of a philosophical point – that way we can change the internals of ManifoldsBase without breaking stuff here.

Reply:
I see. I would prefer less duplication but we can keep it this way too.

Reply:
I do see positive points in both approaches. Mine is a bit more independent of internal things of another package – yours avoids a few lines of code duplication. But thanks that we can leave it like this for now.

If we come across a good reason to have these few lines not double, we can also change it later.